src/shared/logs/collections.py

```diff
@@ -0,0 +1,51 @@
+from collections.abc import Iterator
+
+from shared.logs.raw_events import RawEventType
+
+
+class EventCollection:
+    """
+    A collection of raw events that provides filtering and manipulation capabilities.
+    This class is agnostic to the source of events (suggestions, users, etc.)
+    """
+
+    def __init__(self, events: list[RawEventType], _skip_sort: bool = False) -> None:
+        """Initialize with a list of raw events, optionally skipping sort if already sorted."""
+        if _skip_sort:
+            self._events = events
+        else:
+            self._events = sorted(events, key=lambda event: event.timestamp)
+
+    @property
+    def events(self) -> list[RawEventType]:
+        """Get the current list of events."""
+        return self._events.copy()
```

Comment:
Side note: the underlying list elements are still mutable, so if used carelessly this stays a footgun. I doubt it's worth going full pedantic mode to prevent this. Also since it's an internal interface I would even doubt it's worth being very OOPy here: we could just as well use the class as a means of code organisation, and simply store the list of `RawEvent` and have methods that will return it filtered for various purposes.

Reply:
Done, now collections of events (raw events or folded events) no longer have a dedicated class but are simple lists.